types/chrome/index.d.ts

```diff
@@ -6742,6 +6742,159 @@ declare namespace chrome.printerProvider {
+        isDefault: boolean;
+        /** The name of the printer. */
+        name: string;
+        /**
+         * The value showing how recent the printer was used for printing from Chrome.
+         * The lower the value is the more recent the printer was used.
+         * The minimum value is 0.
+         * Missing value indicates that the printer wasn't used recently.
+         * This value is guaranteed to be unique amongst printers.
+         */
+        recentlyUsedRank?: number;
+        /** The source of the printer (user or policy configured). */
+        source: PrinterSource;
+        /** The printer URI. This can be used by extensions to choose the printer for the user. */
+        uri: string;
+    }
+
+    /** The source of the printer. */
+    export enum PrinterSource {
+        /** Printer was added by user. */
+        USER = "USER",
+        /** Printer was added via policy. */
+        POLICY = "POLICY",
+    }
+
+    /** The status of the printer. */
+    export enum PrinterStatus {
+        /** The door of the printer is open. Printer still accepts print jobs. */
+        DOOR_OPEN = "DOOR_OPEN",
+        /** The tray of the printer is missing. Printer still accepts print jobs. */
+        TRAY_MISSING = "TRAY_MISSING",
+        /** The printer is out of ink. Printer still accepts print jobs. */
+        OUT_OF_INK = "OUT_OF_INK",
+        /** The printer is out of paper. Printer still accepts print jobs. */
+        OUT_OF_PAPER = "OUT_OF_PAPER",
+        /** The output area of the printer (e.g. tray) is full. Printer still accepts print jobs. */
+        OUTPUT_FULL = "OUTPUT_FULL",
+        /** The printer has a paper jam. Printer still accepts print jobs. */
+        PAPER_JAM = "PAPER_JAM",
+        /** Some generic issue. Printer still accepts print jobs. */
+        GENERIC_ISSUE = "GENERIC_ISSUE",
+        /** The printer is stopped and doesn't print but still accepts print jobs. */
+        STOPPED = "STOPPED",
+        /** The printer is unreachable and doesn't accept print jobs. */
+        UNREACHABLE = "UNREACHABLE",
+        /** The SSL certificate is expired. Printer accepts jobs but they fail. */
+        EXPIRED_CERTIFICATE = "EXPIRED_CERTIFICATE",
+        /** The printer is available. */
+        AVAILABLE = "AVAILABLE",
+    }
+
+    export interface SubmitJobRequest {
+        /**
+         * The print job to be submitted.
+         * The only supported content type is "application/pdf", and the Cloud Job Ticket shouldn't include FitToPageTicketItem, PageRangeTicketItem, ReverseOrderTicketItem and VendorTicketItem fields since they are irrelevant for native printing.
+         * All other fields must be present.
+         */
+        job: chrome.printerProvider.PrintJob;
+    }
+
+    export interface SubmitJobResponse {
+        /** The id of created print job. This is a unique identifier among all print jobs on the device. If status is not OK, jobId will be null. */
+        jobId?: string;
```

Comment:
If it's specifically string | null, go ahead and explicitly declare that. Declaring with `TypeName?: T` is a shortcut for `TypeName: T | undefined`, so you'll get an unexpected type here.

Reply:
The documentation misled it with the `optional` tag

After verification, the suggested type is correct :)